packages/auth/auth0-react/src/providers/auth0ProviderWithNavigate.tsx

```diff
@@ -0,0 +1,35 @@
+import { Auth0Provider } from "@auth0/auth0-react";
+import { useNavigate } from "react-router-dom";
+
+import Auth0ProviderWithNavigateProps from "../types/Auth0ProviderWithNavigateProps";
+
+// See https://stackoverflow.com/a/73938067 for using with RouteProvider (v6)
+export const Auth0ProviderWithNavigate = ({
+  children,
+}: Auth0ProviderWithNavigateProps) => {
+  const navigate = useNavigate();
+
+  const domain = import.meta.env.VITE_AUTH0_DOMAIN;
+  const clientId = import.meta.env.VITE_AUTH0_CLIENT_ID;
+  const redirectUri = import.meta.env.VITE_AUTH0_CALLBACK_URL;
+
+  const onRedirectCallback = (appState: any) => {
+    navigate(appState?.returnTo || window.location.pathname);
+  };
+
+  if (!(domain && clientId && redirectUri)) {
+    console.log("Nice goin' Ace");
```

Comment:
This should be removed for a production app but was helpful when debugging why I was seeing just a blank screen.